src/util/codec/mysql/json.rs

```diff
@@ -563,4 +710,33 @@ mod test {
             assert!(resp.is_err());
         }
     }
+
+    #[test]
+    fn test_cmp_json() {
+        let json_null = Json::from_str("null").unwrap();
+        let json_bool_true = Json::from_str("true").unwrap();
+        let json_bool_false = Json::from_str("false").unwrap();
+        let json_integer_big = Json::from_str("5").unwrap();
+        let json_integer_small = Json::from_str("3").unwrap();
+        let json_str_big = Json::from_str(r#""hello, world""#).unwrap();
+        let json_str_small = Json::from_str(r#""hello""#).unwrap();
+        let json_array_big = Json::from_str(r#"["a", "c"]"#).unwrap();
+        let json_array_small = Json::from_str(r#"["a", "b"]"#).unwrap();
+        let json_object = Json::from_str(r#"{"a": "b"}"#).unwrap();
+        let test_cases = vec![
+            (&json_null, &json_integer_small),
+            (&json_integer_small, &json_integer_big),
+            (&json_integer_big, &json_str_small),
+            (&json_str_small, &json_str_big),
+            (&json_str_big, &json_object),
+            (&json_object, &json_array_small),
+            (&json_array_small, &json_array_big),
+            (&json_array_big, &json_bool_false),
+            (&json_bool_false, &json_bool_true),
+        ];
+
+        for (left, right) in test_cases {
```

Comment:
It would be failed with 

```
722 |         for (&left, &right) in test_cases {
    |                     ^-----
    |                     ||
    |                     |hint: to prevent move, use `ref right` or `ref mut right`
    |                     cannot move out of borrowed content

```